ci/common-ci.yml

```diff
@@ -136,3 +136,7 @@ variables:
     strategy: depend
     forward:
       pipeline_variables: true
+  variables:
+    PIKA_SMALL_STACK_SIZE: 0x40000
+    PIKA_MEDIUM_STACK_SIZE: 0x40000
```

Comment:
The defaults are
- small stack size in release: 0x10000 (64 KiB)
- small stack size in debug: 0x20000 (128 KiB)
- medium stack size: 0x20000 (128 KiB)

I'm making them all 256 KiB for now to leave a bit of margin (large is way bigger, so not touching that). If we see segfaults with these sizes it's more likely to be a "regular" segfault, or stack overflow of the more pathological kind (i.e. very deep recursion).

Reply:
Note that I've applied the stack size change to _all_ CI configurations, since I'd like to see if this has any effect on the MC pipelines as well. I'm not quite keeping track of which builds/failures have happened with what fixes, but there have been recent failures on the MC pipelines and/or MC tests as well, and if they remain then it's likely not due to the stack size being too small.